File: C Programs/DAA_Practical/linear_search.c
#include<stdio.h>

void linear_search(int []);
int n;
int key;

int main(){
    printf( "Enter no of elements to be entered : ");
    scanf("%d",&n);
    int arr[n];
    printf("Enter Elements:\n");
    for(int i =0; i<n;i++){
        scanf("%d",&arr[i]);
    }
    linear_search(arr);
    return 0;
}

void linear_search(int arr[]){
    int itr=0;
    printf("Enter Element to Find : ");
    scanf("%d",&key);
    for(int i =0; i<n;i++){
        itr++;
        if(arr[i]==key){
            printf("Element found\n");
            printf("Element found at index %d\n",i);
            printf("Number of iteration used is %d\n",itr);
        }
    }
}

File: Java_Programs/DSA/linear_search.java
package DSA;

public class linear_search {
    public static void main(String[] args) {
        int  nums[]={1,2,3,4,5,6,7,8};

        int result = linearsearch(nums,5);
                if(result == -1){
                    System.out.println("Number Not found");
                }else{
                    System.out.println("Number found at index "+result);
                }
            }
        
        static int linearsearch(int[] nums,int tar){
            for(int i = 0;i<nums.length;i++){
                if(nums[i]==tar){
                 return i;
                }
            }
            return -1;
        }
}
